Add -tmpMnt to change namespace mount point

Lencerf opened the pull request:
Some applications may need to access the original contents in `/tmp` directory of the local host. Adding `-tmpMnt` to allow changing the private namespace mount point to avoid conflict.

Signed-off-by: Changyuan Lyu <changyuanl@google.com>

rminnich commented on session/session.go:
```diff
@@ -102,13 +103,13 @@ func (s *Session) TmpMounts() error {
 	// This ought to be safe as it is a private namespace mount.
 	// (or we are started with a clean namespace in Plan 9).
-	for _, n := range []string{"/tmp/cpu", "/tmp/local", "/tmp/merge", "/tmp/root", "/home"} {
+	for _, n := range []string{s.tmpMnt + "/cpu", s.tmpMnt + "/local", s.tmpMnt + "/merge", s.tmpMnt + "/root", "/home"} {
```
This needs to use filepath.Join, not +

Lencerf replied:
Done.

rminnich commented on session/session_linux.go:
```diff
@@ -96,8 +96,8 @@ func (s *Session) Namespace() (error, error) {
 	}
-	v("CPUD: mount 127.0.0.1 on /tmp/cpu 9p %#x %s", flags, opts)
-	if err := unix.Mount("localhost", "/tmp/cpu", "9p", flags, opts); err != nil {
+	v("CPUD: mount 127.0.0.1 on %s/cpu 9p %#x %s", s.tmpMnt, flags, opts)
```
use filepath.Join(s.tmpMnt, "cpu")

Lencerf replied:
Done

rminnich commented on session/session_linux.go:
```diff
@@ -96,8 +96,8 @@ func (s *Session) Namespace() (error, error) {
-	v("CPUD: mount 127.0.0.1 on /tmp/cpu 9p %#x %s", flags, opts)
-	if err := unix.Mount("localhost", "/tmp/cpu", "9p", flags, opts); err != nil {
+	v("CPUD: mount 127.0.0.1 on %s/cpu 9p %#x %s", s.tmpMnt, flags, opts)
+	if err := unix.Mount("localhost", s.tmpMnt+"/cpu", "9p", flags, opts); err != nil {
```
pretty much everywhere you've to any kind of assembling of file names, use filepath.Join

There's a very small chance cpud could run on non-linux systems some day.

Lencerf replied:
Done.

Lencerf committed:
Add -tmpMnt to change namespace mount point

Some applications may need to access the original contents in /tmp
directory of the local host. Adding -tmpMnt to allow changing the
private namespace mount point to avoid conflict.

Signed-off-by: Changyuan Lyu <changyuanl@google.com>

Lencerf commented:
@rminnich In cmds/cpud/serve.go, since `initsetup()` is called in the `runAsInit` mode, not in the `-remote` mode, do we need the following tmpfs mount?
https://github.com/u-root/cpu/blob/b8d979804e637869b041e1599ab7b03ffdd31557/cmds/cpud/serve.go#L48-L51
 
If it is running as init, we might need that mount if root is on, e.g., readonly media.

rminnich committed:
Merge branch 'main' into tempmountdir
